src/main/java/nl/tudelft/planningstool/database/entities/courses/CourseRelation.java

```diff
@@ -0,0 +1,39 @@
+package nl.tudelft.planningstool.database.entities.courses;
+
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+import nl.tudelft.planningstool.database.entities.User;
+
+import javax.persistence.*;
+import java.io.Serializable;
+
+@Data
+@Entity
+@Table(name = "course_relations")
+@EqualsAndHashCode(of = {
+        "course", "user"
+})
+public class CourseRelation implements Serializable {
```

Comment:
Because of the composite key using multiple `@Id` annotations, this entity needs an [`@IdClass`](http://docs.oracle.com/javaee/5/api/javax/persistence/IdClass.html). After browsing the web for hours I finally found a [decent resource](http://www.kawoolutions.com/Technology/JPA,_Hibernate,_and_Co./JPA_Composite_Key_Variants) on how composite keys should be annotated in JPA. Id classes should apparently be Serializable. Furthermore, they should not point to your relations, but their key (be it id classes or longs). The fields do not have to be annotated, but the attribute names must match to the name in the entity and equals/hashcode should be implemented. I believe `@IdClass` would look like the following (assuming IdClass can be mixed with embedded ID's):

```java
@Data
public static clas CourseRelationId implements Serializable {
     private CourseEdition edition;
     private Integer user;
}
```

Then on the `CourseRelation`:
```java
@IdClass(CourseRelationId.class)
public class CourseRelation implements Serializable {
//...
```

Something similar should be implemented for the `Occurence` entity.